### PR TITLE
Add Quick Sort C++ and improved README

### PR DESCRIPTION
**Pull request by Jaidev2001**
Quicksort program added in c++ language and edited readme with proper describing of algorithm in steps. :smile:

### CODE REVIEW

**Comment by Himanshu40** [comment]
Reference to #12 #3

File: Quick_Sort/quicksort.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    int i;  
+    for (i = 0; i < size; i++)  
+        cout << arr[i] << " ";  
+    cout << endl;  
+}  
+  
+ 
+int main()  
+{  
+    int arr[] = {10, 7, 8, 9, 1, 5};
```
Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:55]:
```suggestion
    
```

Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:55]:
```suggestion
    Take run-time input instead of giving static input
```

Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:55]: Take run-time input rather than static input or hard-coded input.

File: Quick_Sort/quicksort.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}  
+ 
+int partition (int arr[], int low, int high)  //this function takes the last element as pivot and places the pivot at its correct position
+{  
+    int pivot = arr[high]; 
+    int i = (low - 1);   
+  
+    for (int j = low; j <= high - 1; j++)  
+    {  
+        //i++;
```
Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:19]:
```suggestion
```

File: Quick_Sort/quicksort.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if (arr[j] <=pivot)  
+        {  
+            i++; 
+            swap(&arr[i], &arr[j]);  
+        }  
+    }  
+    swap(&arr[i + 1], &arr[high]);  
+    return (i + 1);
```
Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:27]:
```suggestion

    return (i + 1);  
```

File: Quick_Sort/quicksort.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+void quickSort(int arr[], int low, int high)  //this fucntion implements the quicksort
+{  
+    if (low >= high)  
+        return;
+     
+       
+        int pi = partition(arr, low, high);           
+        quickSort(arr, low, pi - 1);  
+        quickSort(arr, pi + 1, high);  
+
```
Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:40]:
```suggestion
```

File: Quick_Sort/quicksort.cpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    swap(&arr[i + 1], &arr[high]);  
+    return (i + 1);  
+}  
+  
+
+void quickSort(int arr[], int low, int high)  //this fucntion implements the quicksort
+{  
+    if (low >= high)  
+        return;
+
```
Comment (Himanshu40) [inline Quick_Sort/quicksort.cpp:35]:
```suggestion
```

**Review by Himanshu40 — CHANGES_REQUESTED** [review]
1. Remove unusual or excess new lines in each function.
2. Take run-time input rather than static or hard-coded input.

**Review by Himanshu40 — CHANGES_REQUESTED** [review]
1. Add a table and you can see github markdown style on how to add a table or for reference you can see my BubbleSort/README.md where I used a table.

**Comment by Himanshu40** [comment]
I have left some comments on your commits. Have a look !!

**Review by Jaidev2001 — COMMENTED** [review]
Hey, i have edited and make the changes that are required.

**Comment by Himanshu40** [comment]
The output of Quick Sort is not expected like the following:-
```
Himanshu@PREDATOR-HELIOS-300 MINGW64 ~/Desktop
$ ./run.exe

        ---QUICK SORT---
Enter the size of the array: 10
Enter the elements of array:
9
8
7
6
5
4
3
2
1
0

Elements in the array: 9 8 7 6 5 4 3 2 1 0
Elements in ascending order:
0 0 1 2 3 4 5 6 7 8
```
Why there is an extra 0 at the start @Jaidev2001 ?

**Review by Jaidev2001 — COMMENTED** [review]
:smile:

**Pull request closed without merging**